heimctl/retention/scanner.go

```diff
@@ -80,3 +84,60 @@ func ExpiredScanLoop(ctx scope.Context, c cluster.Cluster, pb *psql.Backend, int
 		}
 	}
 }
+
+func scanToDelete(ctx scope.Context, c cluster.Cluster, pb *psql.Backend) error {
+	rows, err := pb.DbMap.Select(
+		psql.Room{},
+		"SELECT name, founded_by, retention_days FROM room WHERE retention_days > 0")
+	if err == sql.ErrNoRows {
+		lastDeleteScan.Set(float64(time.Now().Unix()))
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	for _, row := range rows {
+		room, ok := row.(*psql.Room)
+		if !ok {
+			fmt.Printf("error: expected row of type *psql.Room, got %T\n", row)
+			continue
+		}
+		// don't use grace period here- delete as soon as they expire
+		threshold := time.Now().Add(time.Duration(-room.RetentionDays) * 24 * time.Hour).Unix()
```

Comment:
Don't use .Unix() here, otherwise you're passing an int64 to gorp which compares a timestamp (posted) to an integer.